python/pyiceberg/expressions/visitors.py

```diff
@@ -842,3 +842,59 @@ def visit_bound_predicate(self, predicate: BoundPredicate[L]) -> Set[int]:
 
 def extract_field_ids(expr: BooleanExpression) -> Set[int]:
     return visit(expr, _ExpressionFieldIDs())
+
+
+class _RewriteToDNF(BooleanExpressionVisitor[BooleanExpression]):
+    def visit_true(self) -> BooleanExpression:
+        return AlwaysTrue()
+
+    def visit_false(self) -> BooleanExpression:
+        return AlwaysFalse()
+
+    def visit_not(self, child_result: BooleanExpression) -> BooleanExpression:
```

Comment:
@Fokko, can this be handled by running `rewriteNot` before this visitor is applied? That should avoid the need to have so much logic here.

Reply:
I've done some checks, and that works well! Added some more tests and the logic is indeed redundant here. The difference is that we don't have any `Not()` in the output, but that works well 👍🏻